core/src/test/java/kafka/server/ClientMetricsManagerTest.java

```diff
@@ -0,0 +1,951 @@
+
+        assertEquals(Errors.NONE, response.error());
+        assertFalse(instance.terminating());
+        assertEquals(Errors.NONE, instance.lastKnownError());
+    }
+
+    @Test
+    public void testPushTelemetryOnNewServer() throws UnknownHostException {
+        GetTelemetrySubscriptionsRequest subscriptionsRequest = new GetTelemetrySubscriptionsRequest.Builder(
+            new GetTelemetrySubscriptionsRequestData(), true).build();
+
+        GetTelemetrySubscriptionsResponse subscriptionsResponse = clientMetricsManager.processGetTelemetrySubscriptionRequest(
+            subscriptionsRequest, ClientMetricsTestUtils.requestContext(), 0);
+
+        // Create new client metrics manager which simulates a new server as it will not have any
+        // client instance information but request should succeed as subscription id should match
+        // the one with new client instance.
+
+        ClientMetricsManager newClientMetricsManager = new ClientMetricsManager(config, time);
+
+        PushTelemetryRequest request = new PushTelemetryRequest.Builder(
+            new PushTelemetryRequestData()
+                .setClientInstanceId(subscriptionsResponse.data().clientInstanceId())
+                .setSubscriptionId(subscriptionsResponse.data().subscriptionId()), true).build();
+
+        PushTelemetryResponse response = newClientMetricsManager.processPushTelemetryRequest(
+            request, ClientMetricsTestUtils.requestContext(), 0);
+
+        assertEquals(Errors.NONE, response.error());
+    }
+
+    @Test
+    public void testPushTelemetryAfterPushIntervalTime() throws UnknownHostException {
+        clientMetricsManager.updateSubscription("sub-1", ClientMetricsTestUtils.defaultProperties());
+        assertEquals(1, clientMetricsManager.subscriptions().size());
+
+        GetTelemetrySubscriptionsRequest subscriptionsRequest = new GetTelemetrySubscriptionsRequest.Builder(
+            new GetTelemetrySubscriptionsRequestData(), true).build();
+
+        GetTelemetrySubscriptionsResponse subscriptionsResponse = clientMetricsManager.processGetTelemetrySubscriptionRequest(
+            subscriptionsRequest, ClientMetricsTestUtils.requestContext(), 0);
+
+        PushTelemetryRequest request = new Builder(
+            new PushTelemetryRequestData()
+                .setClientInstanceId(subscriptionsResponse.data().clientInstanceId())
+                .setSubscriptionId(subscriptionsResponse.data().subscriptionId())
+                .setCompressionType(CompressionType.NONE.id)
+                .setMetrics("test-data".getBytes(StandardCharsets.UTF_8)), true).build();
+
+        PushTelemetryResponse response = clientMetricsManager.processPushTelemetryRequest(
+            request, ClientMetricsTestUtils.requestContext(), 0);
+
+        assertEquals(Errors.NONE, response.error());
+
+        time.setCurrentTimeMs(time.milliseconds() + ClientMetricsTestUtils.DEFAULT_PUSH_INTERVAL_MS);
```

Comment:
This could just be `time.sleep(ClientMetricsTestUtils.DEFAULT_PUSH_INTERVAL_MS)`. Ditto in a few other places.

Reply:
Done.